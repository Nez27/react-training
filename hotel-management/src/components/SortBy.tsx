import { useSearchParams } from 'react-router-dom';
import Select from './Select';

interface ISortByProps {
  options: {
    value: string;
    label: string;
  }[];
}

const SortBy = ({ options }: ISortByProps) => {
  const [searchParams, setSearchParams] = useSearchParams();
  const sortBy = searchParams.get('sortBy') || '';

  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    searchParams.set('sortBy', event.target.value);
    setSearchParams(searchParams);
  };

  // prettier-ignore
  return (
    <Select 
      options={options} 
      value={sortBy} 
      onChange={handleChange} 
    />
  );
};

export default SortBy;
