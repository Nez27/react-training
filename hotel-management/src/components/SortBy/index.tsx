import { useSearchParams } from 'react-router-dom';
import { ChangeEvent, memo } from 'react';

// Components
import Select from '@component/Select';

interface ISortByProps {
  options: {
    value: string;
    label: string;
  }[];
}

const SortBy = memo(({ options }: ISortByProps) => {
  const [searchParams, setSearchParams] = useSearchParams();
  const sortBy = searchParams.get('sortBy') || '';
  const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
    searchParams.set('sortBy', event.target.value);
    setSearchParams(searchParams);
  };

  return (
    <Select
      options={options}
      value={sortBy}
      onChange={handleChange}
      ariaLabel="Sort"
    />
  );
});

export default SortBy;
