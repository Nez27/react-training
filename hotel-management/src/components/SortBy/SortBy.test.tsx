import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

// Components
import SortBy from '.';

describe('<SortBy.test />', () => {
  const options = [
    {
      label: 'Option 1',
      value: 'Value 1',
    },
    {
      label: 'Option 2',
      value: 'Value 2',
    },
  ];
  const wrapper = renderer.create(
    <BrowserRouter>
      <SortBy options={options} />
    </BrowserRouter>
  );

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
