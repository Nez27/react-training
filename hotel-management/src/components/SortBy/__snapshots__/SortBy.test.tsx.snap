// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortBy.test /> render 1`] = `
<select
  aria-label="Sort"
  className="sc-bdnyFh dqXTRP"
  onChange={[Function]}
  value=""
>
  <option
    value="Value 1"
  >
    Option 1
  </option>
  <option
    value="Value 2"
  >
    Option 2
  </option>
</select>
`;
